internal/v4/api.go

```diff
@@ -205,34 +216,38 @@ func (h *Handler) puttableBaseEntityHandler(get baseEntityHandlerFunc, handlePut
 	})
 }
 
-func (h *Handler) updateBaseEntity(id *charm.Reference, fields map[string]interface{}) error {
+func (h *Handler) updateBaseEntity(id *router.ResolvedURL, fields map[string]interface{}) error {
 	if err := h.store.UpdateBaseEntity(id, bson.D{{"$set", fields}}); err != nil {
 		return errgo.Notef(err, "cannot update base entity %q", id)
 	}
 	return nil
 }
 
-func (h *Handler) updateEntity(id *charm.Reference, fields map[string]interface{}) error {
+func (h *Handler) updateEntity(id *router.ResolvedURL, fields map[string]interface{}) error {
 	err := h.store.UpdateEntity(id, bson.D{{"$set", fields}})
 	if err != nil {
-		return errgo.Notef(err, "cannot update %q", id)
+		return errgo.Notef(err, "cannot update %q", &id.URL)
 	}
 	err = h.store.UpdateSearchFields(id, fields)
 	if err != nil {
-		return errgo.Notef(err, "cannot update %q", id)
+		return errgo.Notef(err, "cannot update %q", &id.URL)
 	}
 	return nil
 }
 
-func (h *Handler) updateSearch(id *charm.Reference, fields map[string]interface{}) error {
+func (h *Handler) updateSearch(id *router.ResolvedURL, fields map[string]interface{}) error {
 	return h.store.UpdateSearch(id)
 }
 
-func (h *Handler) updateSearchBase(id *charm.Reference, fields map[string]interface{}) error {
+// updateSearchBase updates the search records for all entities with
+// the same base URL as the given id.
+func (h *Handler) updateSearchBase(id *router.ResolvedURL, fields map[string]interface{}) error {
```

Comment:
However, we must remove the document in ES with a previous promulgated URL under a different base entity as well. Out of scope of this function, but just a note.